dotnet/src/Connectors/Connectors.Memory.Postgres/PostgresVectorStoreCollectionSqlBuilder.cs

```diff
@@ -0,0 +1,447 @@
+                if (keyProperty != null)
+                {
+                    throw new ArgumentException("Record definition cannot have more than one key property.");
+                }
+                keyProperty = keyProp;
+            }
+            else if (property is VectorStoreRecordDataProperty dataProp)
+            {
+                dataProperties.Add(dataProp);
+            }
+            else if (property is VectorStoreRecordVectorProperty vectorProp)
+            {
+                vectorProperties.Add(vectorProp);
+            }
+            else
+            {
+                throw new NotSupportedException($"Property type {property.GetType().Name} is not supported by this store.");
+            }
+        }
+
+        if (keyProperty == null)
+        {
+            throw new ArgumentException("Record definition must have a key property.");
+        }
+
+        var keyName = keyProperty.StoragePropertyName ?? keyProperty.DataModelPropertyName;
+
+        StringBuilder createTableCommand = new();
+        createTableCommand.AppendLine($"CREATE TABLE {(ifNotExists ? "IF NOT EXISTS " : "")}{schema}.\"{tableName}\" (");
+
+        // Add the key column
+        var keyPgTypeInfo = PostgresVectorStoreRecordPropertyMapping.GetPostgresTypeName(keyProperty.PropertyType);
+        createTableCommand.AppendLine($"    \"{keyName}\" {keyPgTypeInfo.PgType} {(keyPgTypeInfo.IsNullable ? "" : "NOT NULL")},");
+
+        // Add the data columns
+        foreach (var dataProperty in dataProperties)
+        {
+            string columnName = dataProperty.StoragePropertyName ?? dataProperty.DataModelPropertyName;
+            var dataPgTypeInfo = PostgresVectorStoreRecordPropertyMapping.GetPostgresTypeName(dataProperty.PropertyType);
+            createTableCommand.AppendLine($"    \"{columnName}\" {dataPgTypeInfo.PgType} {(dataPgTypeInfo.IsNullable ? "" : "NOT NULL")},");
+        }
+
+        // Add the vector columns
+        foreach (var vectorProperty in vectorProperties)
+        {
+            string columnName = vectorProperty.StoragePropertyName ?? vectorProperty.DataModelPropertyName;
+            var vectorPgTypeInfo = PostgresVectorStoreRecordPropertyMapping.GetPgVectorTypeName(vectorProperty);
+            createTableCommand.AppendLine($"    \"{columnName}\" {vectorPgTypeInfo.PgType} {(vectorPgTypeInfo.IsNullable ? "" : "NOT NULL")},");
+        }
+
+        createTableCommand.AppendLine($"    PRIMARY KEY (\"{keyName}\")");
+
+        createTableCommand.AppendLine(");");
+
+        return new PostgresSqlCommandInfo(commandText: createTableCommand.ToString());
+    }
+
+    /// <inheritdoc />
+    public PostgresSqlCommandInfo BuildCreateVectorIndexCommand(string schema, string tableName, VectorStoreRecordVectorProperty vectorProperty)
+    {
+        var vectorColumnName = vectorProperty.StoragePropertyName ?? vectorProperty.DataModelPropertyName;
+        // Only support creating HNSW index creation through the connector.
```

Comment:
If `vectorProperty.IndexKind` isn't specified i.e. null, we should default to hnsw

Reply:
The pgvector index is not required. If index creation is mandatory, is the user allowed to have a way to not create index?

Reply:
In general, I think if pgvector index is not required, then it should be `null` by default and other value like `HNSW` should be used only if specified by user explicitly. But if `HNSW` index is performing better than the default behavior, we may consider using `HNSW` as a default just to recommend the efficient approach.

> How about setting the default value to HNSW, but allowing the user to set it to Null?

For such cases, I would probably introduce `IndexKind.None` instead of null value, since setting this parameter as `null` may be interpreted differently.

Reply:
Changed to default to HNSW if not supplied. I added and `IndexKind.None`, and if that is set, then no index is created.

Reply:
`Flat` is what other dbs are typically calling no-indexing.  Although Azure AI Search calls it Exhaustive K nearest neighbor.  It essentially means a brute force search.
Would it make sense to use the same terminology here too?

See e.g.
https://milvus.io/docs/index.md?tab=floating
https://weaviate.io/developers/weaviate/config-refs/schema/vector-index#flat-indexes
https://redis.io/docs/latest/develop/interact/search-and-query/advanced-concepts/vectors/#flat-index

Reply:
IMO `IndexKind.None` makes sense and is readable to a wide audience, whereas "Flat" would have a number of devs looking at docstrings or asking copilot what a flat index is, and "exhaustive" is more a search concept than an index creation concept.

Reply:
The problem is now we have two names for the same thing.  Flat and None, which is also confusing.  I think it's better to have just the one.

Reply:
Makes sense, moved to using IndexKind.Flat and removed IndexKind.None

Reply:
The pgvector has another index type, `IVFFlat`. The `IVFFlat` index type is not suitable for creating in an empty table, it requires some data to exist in the table.

https://github.com/pgvector/pgvector?tab=readme-ov-file#ivfflat

I'm not sure if IndexKind.Flat with it's similar name would cause any misunderstanding among users?